R/pkg/R/DataFrame.R

```diff
@@ -1177,11 +1177,65 @@ setMethod("dim",
 setMethod("collect",
           signature(x = "SparkDataFrame"),
           function(x, stringsAsFactors = FALSE) {
+            useArrow <- FALSE
```

Comment:
is it possible to factor this out into a internal helper? `"collect"` is getting long (and some of these structType checks are repeated)

Reply:
Yes, in particular for such checking logics, I am going to write a helper function and deduplicate all of them when I finish `dapply` which will be very soon. Would it that make sense as well to you? If you think it should be done here, I can do it here too.